### ISSUE TITLE
lu(Symmetric(Diagonal(...))) fails

### ISSUE THREAD

**Issue by tpapp** [JuliaLang/julia#44877]
```julia
julia> VERSION
v"1.8.0-beta3"

julia> using LinearAlgebra

julia> M = Symmetric(Diagonal(ones(3)))
3×3 Symmetric{Float64, Diagonal{Float64, Vector{Float64}}}:
 1.0  0.0  0.0
 0.0  1.0  0.0
 0.0  0.0  1.0

julia> lu(M)
ERROR: MethodError: no method matching lu!(::Diagonal{Float64, Vector{Float64}}, ::RowMaximum; check=true)
```

**Comment by tpapp** [comment]
Suggestion:

```julia
Symmetric(D::Diagonal, uplo=:U) = D
```
and the same for `Hermitian`. Possibly checking `uplo`.

The downside is violating `T(...) isa T`.

**Comment by dkarrasch** [comment]
One would need to respect the array of arrays case:

```julia
Symmetric(D::Diagonal, uplo=:U) = Diagonal(symmetric.(D.diag), uplo)
```

Or pick up some of https://github.com/JuliaLang/julia/pull/41288.

**Comment by tpapp** [comment]
I don't understand why #41288 was closed in the first place.

The issue is not explicitly why LU is useful for `Diagonal` in the first place (it isn't), or why `Symmetric(Diagonal(...))` makes sense (it doesn't, really). But code that wants to handle a generic `AbstractMatrix` can end up with constructs like this so there should be code path that just works.

@andreasnoack, I would appreciate your thoughts on this.

**Comment by dkarrasch** [comment]
This has been fixed, perhaps in #47107. The question is whether the resulting `LU` object is useful. Getting properties à la `F.L` creates dense factors, while `F.factors` is a memory-efficient `Diagonal`. Some further optimizations are proposed in #48189.
